<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerStenosisMeasurement3DModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerStenosisMeasurement3DModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>371</width>
    <height>685</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="parameterSetFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="parameterSetLabel">
       <property name="text">
        <string>Parameter set:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="qMRMLNodeComboBox" name="parameterSetSelector">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>A parameter set groups parameters that define a named study distinctly.</string>
       </property>
       <property name="nodeTypes">
        <stringlist notr="true">
         <string>vtkMRMLStenosisMeasurement3DParameterNode</string>
        </stringlist>
       </property>
       <property name="showHidden">
        <bool>true</bool>
       </property>
       <property name="hideChildNodeTypes">
        <stringlist notr="true"/>
       </property>
       <property name="baseName">
        <string>StenosisMeasurement3D</string>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
       <property name="interactionNodeSingletonTag">
        <string notr="true"/>
       </property>
       <property name="selectNodeUponCreation">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="inputShapeLabel">
        <property name="text">
         <string>Wall surface:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="inputShapeSelector">
        <property name="toolTip">
         <string>Select an input shape (tube) node, drawn to represent the vascular wall.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLMarkupsShapeNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string>Tube</string>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="selectNodeUponCreation">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="inputFiducialLabel">
        <property name="text">
         <string>Boundary node:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="inputFiducialSelector">
        <property name="toolTip">
         <string>Select an input markups fiducial node.

The first and second points are the boundaries between which the analysis will be performed.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLMarkupsFiducialNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string>P</string>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="selectNodeUponCreation">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="inputSegmentLabel">
        <property name="text">
         <string>Lumen surface:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLSegmentSelectorWidget" name="inputSegmentSelector">
        <property name="toolTip">
         <string>Select a segment representing the vascular lumen.

This should ideally exceed the wall surface a little.</string>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="selectNodeUponCreation">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="regionsHorizontalLayout">
        <item>
         <widget class="QLabel" name="regionInfoLabel">
          <property name="toolTip">
           <string>Ideally, there should be only one region in the input segment.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="fixRegionToolButton">
          <property name="toolTip">
           <string>If the segment has more than one region, this function replaces the input segment by a new segment having a single region. It is created from the largest region of the input segment.

The result must be inspected before use. If it is not satisfactory, the source segment should be updated by any other means.</string>
          </property>
          <property name="text">
           <string>Fix</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="outputCollapsibleButton">
     <property name="toolTip">
      <string/>
     </property>
     <property name="text">
      <string>Outputs</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QFormLayout" name="resultFormLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="wallLabel">
          <property name="text">
           <string>Wall:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="wallResultLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="lumenLabel">
          <property name="text">
           <string>Lumen:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="lumenResultLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="lesionLabel">
          <property name="text">
           <string>Lesion:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="lesionResultLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="stenosisLabel">
          <property name="text">
           <string>Stenosis:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="stenosisResultLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="lengthLabel">
          <property name="text">
           <string>Length:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="lengthResultLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="ctkCollapsibleButton" name="modelCollapsibleButton">
        <property name="text">
         <string>Model</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QFormLayout" name="modelFormLayout">
           <item row="1" column="0">
            <widget class="QLabel" name="lesionModelLabel">
             <property name="text">
              <string>Lesion:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="qMRMLNodeComboBox" name="lesionModelSelector">
             <property name="toolTip">
              <string>Show the lesion between the boundary points as a model.</string>
             </property>
             <property name="nodeTypes">
              <stringlist notr="true">
               <string>vtkMRMLModelNode</string>
              </stringlist>
             </property>
             <property name="baseName">
              <string>Lesion</string>
             </property>
             <property name="noneEnabled">
              <bool>true</bool>
             </property>
             <property name="editEnabled">
              <bool>true</bool>
             </property>
             <property name="renameEnabled">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="outputTableLabel">
             <property name="text">
              <string>Table:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="outputTableHorizontalLayout">
             <item>
              <widget class="qMRMLNodeComboBox" name="outputTableSelector">
               <property name="toolTip">
                <string>Specify a table to store the result in append mode.</string>
               </property>
               <property name="nodeTypes">
                <stringlist notr="true">
                 <string>vtkMRMLTableNode</string>
                </stringlist>
               </property>
               <property name="hideChildNodeTypes">
                <stringlist notr="true"/>
               </property>
               <property name="baseName">
                <string>Result table</string>
               </property>
               <property name="noneEnabled">
                <bool>true</bool>
               </property>
               <property name="editEnabled">
                <bool>true</bool>
               </property>
               <property name="renameEnabled">
                <bool>true</bool>
               </property>
               <property name="interactionNodeSingletonTag">
                <string notr="true"/>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="updateBoundaryPointsSpinBox">
               <property name="toolTip">
                <string>Move the boundary points to the positions referenced in this table row.

Note: this is intended for processing again at known locations. The study and the results stored in the table are not taken into consideration.</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="ctkMenuButton" name="applyButton">
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkMenuButton</class>
   <extends>QPushButton</extends>
   <header>ctkMenuButton.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSegmentSelectorWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLSegmentSelectorWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>lesionModelSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>457</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputSegmentSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputFiducialSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputShapeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputTableSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>463</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerStenosisMeasurement3DModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>parameterSetSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
